lib/awful/screenshot.lua

```diff
@@ -0,0 +1,884 @@
+---------------------------------------------------------------------------
+--- Screenshots and related configuration settings
+--
+-- @author Brian Sobulefsky &lt;brian.sobulefsky@protonmail.com&gt;
+-- @copyright 2021 Brian Sobulefsky
+-- @inputmodule awful.screenshot
+---------------------------------------------------------------------------
+
+-- Grab environment we need
+local capi = { root = root,
+               screen = screen,
+               client = client,
+               mousegrabber = mousegrabber }
+local aw_screen = require("awful.screen")
+local gears = require("gears")
+local beautiful = require("beautiful")
+local wibox = require("wibox")
+local cairo = require("lgi").cairo
+local naughty = require("naughty")
+
+-- The module to be returned
+local module = { mt = {} }
+-- The screenshow object created by a call to the screenshow module
+local screenshot = {}
+-- The various methods of taking a screenshow (root window, client, etc).
+local screenshot_methods = {}
+
+-- Configuration data
+local ss_dir = nil
+local ss_prfx = nil
+local frame_color = gears.color("#000000")
+local initialized = nil
+
+
+-- Internal function to get the default screenshot directory
+-- 
+-- Can be expanded to read X environment variables or configuration files.
+local function get_default_dir()
+  -- This can be expanded
+  local d = os.getenv("HOME")
+  
+  if d then
+    d = string.gsub(d, '/*$', '/') .. 'Images/'
```

Comment:
read XDG spec about images' dir

Reply:
I found [this on Archlinux](https://wiki.archlinux.org/title/XDG_user_directories) on the user directories. How do you want this handled for the screenshot module? It looks like it would be in '~/.config/user-dirs.dirs'. I say in the comment we can expand it to include parsing such a file. Is there already a gears routine to get this?

Reply:
see our gears.filesystem please for similar functions (or mb even better add them there, not here)

Reply:
why do you trying to find XDG spec on arch linux website? here is not only explained everything, but even given example (very similar to our gears, again):

https://www.freedesktop.org/wiki/Software/xdg-user-dirs/

Reply:
i believe the most practical way would be to parse that file and replace $HOME to currently detected home dir

this is still not very robust - but would avoid making code asynchronous for external call of bash